packages/roosterjs-content-model-plugins/lib/autoFormat/AutoFormatPlugin.ts

```diff
@@ -112,14 +119,23 @@ export class AutoFormatPlugin implements EditorPlugin {
 
     private handleEditorInputEvent(editor: IEditor, event: EditorInputEvent) {
         const rawEvent = event.rawEvent;
-        if (rawEvent.inputType === 'insertText') {
+        const selection = editor.getDOMSelection();
+        if (
+            rawEvent.inputType === 'insertText' &&
+            selection &&
+            selection.type === 'range' &&
+            selection.range.collapsed
+        ) {
             switch (rawEvent.data) {
                 case ' ':
-                    const { autoBullet, autoNumbering, autoLink } = this.options;
+                    const { autoBullet, autoNumbering, autoLink, autoHyphen } = this.options;
                     keyboardListTrigger(editor, autoBullet, autoNumbering);
                     if (autoLink) {
                         createLinkAfterSpace(editor);
                     }
+                    if (autoHyphen) {
+                        transformHyphen(editor);
+                    }
```

Comment:
Can we call `formatTextSegmentBeforeSelectionMarker` here, then call `keyboardListTrigger`, `createLinkAfterSpace`, `transformHyphen` inside it? So that we only need to format with Content Model once.

Reply:
I created a separated PR: https://github.com/microsoft/roosterjs/pull/2557 for it, since I needed to change a lot of files.